swarm/storage/mru/resource.go

```diff
@@ -245,45 +238,55 @@ func (h *Handler) SetStore(store *storage.NetStore) {
 // It implements the storage.ChunkValidator interface
 func (h *Handler) Validate(addr storage.Address, data []byte) bool {
 
+	if data[0] == 0 && data[1] == 0 && len(data) > common.AddressLength {
+		//metadata chunk
+		rootAddr, _ := metadataHash(data)
+		valid := bytes.Equal(addr, rootAddr)
+		if !valid {
+			log.Warn("Invalid root metadata chunk")
+		}
+		return valid
+	}
+
 	// does the chunk data make sense?
 	// This is not 100% safe evaluation, since content addressed data can coincidentally produce data with length header matching content size
-	signature, period, version, name, parseddata, _, err := h.parseUpdate(data)
+	mru, err := parseUpdate(data)
 	if err != nil {
 		log.Warn("Invalid resource chunk")
 		return false
 	}
 
 	// Check if signature is valid against the chunk address
 	// Since we force signatures to be used, we can know be sure that this is valid data
-	digest := h.keyDataHash(addr, parseddata)
-	publicKey, err := crypto.SigToPub(digest.Bytes(), signature[:])
+	ownerAddr, err := mru.Verify(addr)
 	if err != nil {
 		log.Warn("Unparseable signature in resource chunk %s", addr)
 		return false
 	}
-	signAddr := crypto.PubkeyToAddress(*publicKey)
-	nameHash := ens.EnsNode(name)
-	checkAddr := h.resourceHash(period, version, nameHash, signAddr)
-	if !bytes.Equal(checkAddr, addr) {
-		log.Warn("Invalid signature on resource chunk")
+
+	// Check if who signed the resource update really owns the resource
+	if !verifyResourceOwnership(ownerAddr, mru.metaHash, mru.rootAddr) {
+		log.Warn("Update signer does not match with resource owner")
```

Comment:
Ascertain ownership of the resource without having to lookup or query ENS.

In the later PR Steps, this is refactored so `parseUpdate` returns an error if all these checks fail, further simplifying the `Validate` handler.